parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -486,7 +488,12 @@ sqliteCodebase debugName root localOrRemote action = do
               branchHashLength = branchHashLength,
               branchHashesByPrefix = branchHashesByPrefix,
               lcaImpl = (Just sqlLca),
-              beforeImpl = (Just \l r -> Sqlite.runTransaction conn $ fromJust <$> CodebaseOps.before l r)
+              beforeImpl = (Just \l r -> Sqlite.runTransaction conn $ fromJust <$> CodebaseOps.before l r),
+              namesWithinPath = \path -> Sqlite.runReadOnlyTransaction conn \runTx ->
+                runTx (CodebaseOps.namesWithinPath path),
+              updateNameLookup = Sqlite.runWriteTransaction conn $ \runTx -> do
```

Comment:
I think `runTransaction` would work better here, for readability - `runReadOnlyTransaction` / `runWriteTransaction` are necessary when juggling two different connections.